### PR TITLE
[RLlib] Examples folder: All `training_iteration` translations.

### PR DESCRIPTION
**Pull request by sven1977**
All scripts in rllib/examples that use `execution_plan` are translated to using the new `training_iteration` API.
* random_parametric_agent.py
* two_trainer_workflow.py

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/execution/train_ops.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
 
 @ExperimentalAPI
-def train_one_step(trainer, train_batch) -> Dict:
+def train_one_step(trainer, train_batch, policies_to_train=None) -> Dict:
```
Comment (avnishn) [inline rllib/execution/train_ops.py:39]: Where is this change implemented in the code base?

File: rllib/examples/two_trainer_workflow.py [hunk 2]
```diff
@@ -10,128 +10,146 @@
...
+
+                ppo_batches.append(ppo_batch)
+                num_env_steps += ppo_batch.count
 
-        def add_dqn_metrics(batch):
+        # DQN sub-flow.
+        dqn_train_results = {}
+        dqn_train_batch = self.local_replay_buffer.replay()
+        if dqn_train_batch is not None:
+            dqn_train_results = train_one_step(self, dqn_train_batch, ["dqn_policy"])
```
Comment (avnishn) [inline rllib/examples/two_trainer_workflow.py:114]: Oh it's used here.

**Review by avnishn — APPROVED** [review]
Looks good to me, have some questions.

File: rllib/evaluation/rollout_worker.py [hunk 3]
```diff
@@ -922,8 +922,9 @@ def learn_on_batch(self, samples: SampleBatchType) -> Dict:
                     summarize(samples)
                 )
             )
+
+        info_out = {}
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/evaluation/rollout_worker.py:926]: This was a bug. There are some cases, where `info_out` would not be defined later in the code. I wanted to make sure it's at least an empty dict :)

File: rllib/execution/train_ops.py (hunk 1 above)
Comment (sven1977) [reply rllib/execution/train_ops.py:39]: Yeah, actually, let me add a nice docstring for this function. ...

File: rllib/evaluation/rollout_worker.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (gjoliver) [reply rllib/evaluation/rollout_worker.py:926]: yeah, it's a good idea to define info_out here.
but now that we are doing this, we should try to make sure we always and only do info_out.update(...) in the subsequent code, like line 945.

File: rllib/examples/random_parametric_agent.py [hunk 4]
```diff
@@ -64,29 +66,31 @@ def set_weights(self, weights):
         pass
 
 
-# Create a new Trainer using the Policy and config defined above and a new
-# execution plan.
+# Backward compatibility, just in case users want to use the erroneous old name.
+RandomParametriclPolicy = RandomParametricPolicy
```
Comment (gjoliver) [inline rllib/examples/random_parametric_agent.py:70]: is this really necessary?? :) :)
this is an example script ... folks probably shouldn't import an example script and use it as a library?

File: rllib/examples/random_parametric_agent.py [hunk 5]
```diff
@@ -64,29 +66,31 @@ def set_weights(self, weights):
...
-        )
+    @override(Trainer)
+    def training_iteration(self) -> ResultDict:
+        # Perform rollouts (only for collecting metrics later).
+        _ = synchronous_parallel_sample(self.workers)
 
-        # Return training metrics.
-        return StandardMetricsReporting(rollouts, workers, config)
+        # Return (empty) training metrics.
+        return {}
```
Comment (gjoliver) [inline rllib/examples/random_parametric_agent.py:93]: why not collect rollout related metrics here?

File: rllib/examples/two_trainer_workflow.py [hunk 6]
```diff
@@ -10,128 +10,146 @@
...
+            ppo_train_batch.agent_steps(),
+        )
+        ppo_train_batch = MultiAgentBatch(
+            {"ppo_policy": ppo_train_batch}, ppo_train_batch.count
         )
+        ppo_train_results = train_one_step(self, ppo_train_batch, ["ppo_policy"])
 
-        return StandardMetricsReporting(train_op, workers, config)
+        # Combine results for PPO and DQN into one results dict.
+        results = dict(ppo_train_results, **dqn_train_results)
```
Comment (gjoliver) [inline rllib/examples/two_trainer_workflow.py:151]: this small example is actually really cool!! 2 trainers train 2 policies for different agents in a same ma env. nice.

it doesn't feel exactly right here though, we should be combining the result dict? for example, not overwriting ppo steps with dqn steps, but summing them up?
I wonder how Concurrently does it.

**Review by gjoliver — COMMENTED** [review]
nice pr. I know it's already merged, but I have a few comments ...

File: rllib/examples/random_parametric_agent.py (hunk 5 above)
Comment (sven1977) [reply rllib/examples/random_parametric_agent.py:93]: B/c it's done automatically by RLlib after this. So we always just return the learner stats here.
But yes, we should start thinking about a way to customize this bit of the iteration.

File: rllib/examples/two_trainer_workflow.py (hunk 6 above)
Comment (sven1977) [reply rllib/examples/two_trainer_workflow.py:151]: No, they are not overridden. The PPO stats and DQN stats will reside under their policy ID keys.

```
results = {
"ppo_policy": [some stats],
"dqn_policy": [some other stats]
}
```

**Pull request merged**